sonar-db/src/main/java/org/sonar/db/permission/PermissionQuery.java

```diff
@@ -19,106 +19,98 @@
+import static org.sonar.api.utils.Paging.offset;
+import static org.sonar.db.DatabaseUtils.buildLikeValue;
 
 /**
- * Query used to get users and groups from a permission
+ * Query used to get users and groups permissions
  */
+@Immutable
 public class PermissionQuery {
-
+  public static final int RESULTS_MAX_SIZE = 100;
+  public static final int SEARCH_QUERY_MIN_LENGTH = 3;
+  public static final int DEFAULT_PAGE_SIZE = 20;
   public static final int DEFAULT_PAGE_INDEX = 1;
-  public static final int DEFAULT_PAGE_SIZE = 100;
-
-  public static final String ANY = "ANY";
-  public static final String IN = "IN";
-  public static final String OUT = "OUT";
-  public static final Set<String> AVAILABLE_MEMBERSHIP = ImmutableSet.of(ANY, IN, OUT);
 
   private final String permission;
-  private final String component;
+  private final String componentUuid;
   private final String template;
-  private final String membership;
-  private final String search;
+  private final String searchQuery;
+  private final String searchQueryToSql;
+  private final boolean withPermissionOnly;
+  private final List<String> logins;
+  private final List<String> groupNames;
 
-  // for internal use in MyBatis
-  final String searchSql;
-
-  // max results per page
   private final int pageSize;
-  // index of selected page. Start with 1.
-  private final int pageIndex;
-  // offset. Starts with 0.
   private final int pageOffset;
 
   private PermissionQuery(Builder builder) {
     this.permission = builder.permission;
-    this.component = builder.component;
+    this.withPermissionOnly = builder.withPermissionOnly;
+    this.componentUuid = builder.componentUuid;
     this.template = builder.template;
-    this.membership = builder.membership;
-    this.search = builder.search;
-    this.searchSql = searchToSql(search);
-
+    this.searchQuery = builder.searchQuery;
+    this.searchQueryToSql = builder.searchQuery == null ? null : buildLikeValue(builder.searchQuery, WildcardPosition.BEFORE_AND_AFTER).toLowerCase(Locale.ENGLISH);
     this.pageSize = builder.pageSize;
-    this.pageIndex = builder.pageIndex;
-    this.pageOffset = (builder.pageIndex - 1) * builder.pageSize;
+    this.pageOffset = offset(builder.pageIndex, builder.pageSize);
+    this.logins = builder.logins == null ? Collections.emptyList() : ImmutableList.copyOf(builder.logins);
+    this.groupNames = builder.groupNames == null ? Collections.emptyList() : ImmutableList.copyOf(builder.groupNames);
   }
 
-  private static String searchToSql(@Nullable String s) {
-    String sql = null;
-    if (s != null) {
-      sql = StringUtils.replace(StringUtils.upperCase(s), "%", "/%");
-      sql = StringUtils.replace(sql, "_", "/_");
-      sql = "%" + sql + "%";
-    }
-    return sql;
+  @CheckForNull
+  public String getPermission() {
+    return permission;
   }
 
-  public String permission() {
-    return permission;
+  public boolean withPermissionOnly() {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)